### ISSUE TITLE
AWS: Network Load Balancer ignores annotations upon creation

### ISSUE THREAD

**Issue by alx13** [kubernetes/kubernetes#82595]
**What happened**:
Code for handling annotations for load balancer is executed only upon load balancer modification, not creation in all versions including master branch.

So, for example, it's not possible to enable cross-zone load balancing upon creation. Only create-then-update process works.
- [1.14.6](https://github.com/kubernetes/kubernetes/blob/v1.14.6/pkg/cloudprovider/providers/aws/aws_loadbalancer.go#L344)
- [1.15.3](https://github.com/kubernetes/kubernetes/blob/v1.15.3/staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go#L326)
- [master](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go#L336)

**What you expected to happen**:
Annotations  should be parsed both upon creation and update of load balancer

Code, that parses annotations should be moved outside `else` block
- [1.14.6](https://github.com/kubernetes/kubernetes/blob/v1.14.6/pkg/cloudprovider/providers/aws/aws_loadbalancer.go#L179)
- [1.15.3](https://github.com/kubernetes/kubernetes/blob/v1.15.3/staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go#L164)
- [master](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go#L174)

or duplicated in `if` block

**How to reproduce it (as minimally and precisely as possible)**:
1. create `Service.spec.type.LoadBalancer` with:
```yaml
annotations:
  service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
  service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
```
 cross-zone load balancing will be off

2. modify something in `Service` definition — CZLB will be on

**Environment**:
- Kubernetes version (use `kubectl version`): 1.14.0—1.14.6, 1.15.0—1.15.3, master
- Cloud provider or hardware configuration: AWS

**Comment by jwm** [comment]
I experienced this recently with 1.14.9 on EKS. A newly created NLB ignored `service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled`.

**Comment by ckdarby** [comment]
Experienced this with EKS 1.15 on EKS.

```YAML
apiVersion: v1
kind: Service
metadata:
  name: {{ template "tensorflow-serving.fullname" . }}
  labels:
    heritage: {{ .Release.Service | quote }}
    release: {{ .Release.Name | quote }}
    chart: {{ template "tensorflow-serving.chart" . }}
    app: {{ template "tensorflow-serving.name" . }}
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
    service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-healthy-threshold: "2"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-interval: "10"
    
spec:
  type: {{ .Values.serviceType }}
  ports:
    - name: grpc-service-port
      port: {{ .Values.grpcServicePort }}
      targetPort: grpc-port
    - name: http-service-port
      port: {{ .Values.restServicePort }}
      targetPort: http-port
  selector:
    release: {{ .Release.Name | quote }}
    app: {{ template "tensorflow-serving.name" . }}
```

**Comment by micahlmartin** [comment]
Can confirm this also appears on EKS 1.16

**Comment by jclangst** [comment]
Can confirm this appears where using k8s 1.16 on KOPS. It would be phenomenal to get cross-zone load balancing and healthcheck configuration annotations available for the NLB load balancer types. Related: #86624

**Comment by jclangst** [comment]
For anyone stumbling on this same issue and trying to ensure high-availability before the annotation features are implemented, I have confirmed that you can enable cross-zone load balancing manually via the AWS console and it appears to remain resilient to future service changes. Additionally, I have done some preliminary testing with the passive health checks for NLBs and it appears clusters can remain resilient to pod failures (assuming redundant pods are available) due to the passive health checking built-in to NLBs.

**Comment by kishorj** [comment]
The health check fix merged in #94546 does not address this annotation when NLB is created
  service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
The annotation is applied only during edits. This needs to be fixed, please reopen this issue.

**Comment by TBBle** [comment]
Since (looking at the [AWS SDK docs](https://docs.aws.amazon.com/sdk-for-go/api/service/elbv2/)) there is nowhere to include attributes in the [`c.elb.CreateLoadBalancer`](https://github.com/kubernetes/kubernetes/blob/ef33126cc7a60da6799821a49cc9772479e86cf7/staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go#L178)'s [`CreateLoadBalancerInput`](https://docs.aws.amazon.com/sdk-for-go/api/service/elbv2/#CreateLoadBalancerInput) parameter, I guess this just requires calling `reconcileLBAttributes` after the `CreateLoadBalancer` call, as is done after the `// sync mappings` block in the modification path.

That suggests that this bug also affects the other [S3 Access Logs Attributes](https://github.com/kubernetes/kubernetes/blob/ef33126cc7a60da6799821a49cc9772479e86cf7/staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go#L52-L54), too.

**Comment by Chili-Man** [comment]
I'm also running into this issue as well, I don't know why this issue has been closed.

**Comment by TBBle** [comment]
It was closed because it was inaccurately marked as fixed by #94546. Hopefully #95247 will land relatively quickly, and make it into the 1.20 release.
